### PR TITLE
Add `no-todo-implementation` rule (fixes #64)

### LINKED ISSUE #64: `no-todo-implementation` rule
Disallow having a function (or simply more than 1 argument) in a `test.todo`

See https://github.com/sindresorhus/ava/issues/697

### PR DESCRIPTION
**Pull request by jfmengels**
Add `no-todo-implementation` rule (fixes #64)

### CODE REVIEW

File: test/no-todo-implementation.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const header = `const test = require('ava');\n`;
+
+ruleTester.run('no-todo-implementation', rule, {
+	valid: [
+		header + 'test(t => {});',
+		header + 'test("title", t => {});',
+		header + 'test.todo("title");',
+		header + 'test.todo.cb("title");',
+		header + 'notTest.todo(t => {});',
+			// shouldn't be triggered since it's not a test file
```
Comment (sindresorhus) [inline test/no-todo-implementation.js]: Incorrect indent

File: rules/no-todo-implementation.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const functionExpressions = [
+	'FunctionExpression',
+	'ArrowFunctionExpression'
+];
+
+function isFunction(node) {
+	return node &&
+		functionExpressions.indexOf(node.type) !== -1;
+}
```
Comment (sindresorhus) [inline rules/no-todo-implementation.js]: Can you move this (line 5-13) to https://github.com/avajs/eslint-plugin-ava/blob/master/util.js ?

File: docs/rules/no-todo-implementation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Ensure `test.todo()` is not given an implementation function
+
+Translations: [Français](https://github.com/avajs/ava-docs/blob/master/fr_FR/related/eslint-plugin-ava/docs/rules/no-todo-implementation.md)
+
+`test.todo` is a method that AVA proposes to enumerate test cases that have yet to be implemented. It is not meant to be passed a function to implement the test, and if given one, AVA will throw an error. If you added an implementation, then you are probably meaning to remove the `todo` modifier.
```
Comment (sindresorhus) [inline docs/rules/no-todo-implementation.md]: Made it a bit shorter:

```
[`test.todo()`](https://github.com/avajs/ava#test-placeholders-todo) is intended for planning tests. It's not meant to be passed a function to implement the test, and if given one, AVA will throw an error. If you added an implementation, you probably meant to remove the `todo` modifier.
```

**Comment by sindresorhus** [comment]
I see you've been productive today 🤗

**Comment by jfmengels** [comment]
Updated. And yeah, I had a long train trip on friday, so I worked on this. Will see what I can work on on the way back later tonight ;)

**Comment by sindresorhus** [comment]
I love trains for coding. I'm always productive there.

**Comment by sindresorhus** [comment]
This looks great! Thank you :)

**Comment by jfmengels** [comment]
> I love trains for coding. I'm always productive there.

Me too. This one moved a lot so I had to stop at certain points in order to not get sick :sweat_smile:. And no problem. It has been a while since I did a lot of work here.

**Pull request merged**